File: tiq-e/others/4/main.py
from time import perf_counter
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        if not numRows:
            return []
        ans = [[1]]
        for i in range(1, numRows):
            li = [1]
            for j in range(len(ans[-1]) - 1):
                li.append(ans[-1][j] + ans[-1][j + 1])
            li.append(1)
            ans.append(li)
        return ans


x = 5

start = perf_counter()
result = Solution().generate(x)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/others/4/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let mut res = Vec::new();
        for x in 0..num_rows as usize {
            res.push(vec![1; x + 1]);
            for y in 1..x {
                res[x][y] = res[x - 1][y - 1] + res[x - 1][y];
            }
        }
        return res;
    }
}

fn main() {
    let x = 5;

    let start = Instant::now();
    let result = Solution::generate(x);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
